File: backend/DTO/Request/ProductRequest.cs
using System.ComponentModel.DataAnnotations;

namespace backend.DTO.Request
{
    public class ProductRequest
    {
        [Required, MaxLength(100)]
        public required string Model { get; set; }

        [Required, MaxLength(255)]
        public required string ProductLink { get; set; }

        [MaxLength(255)]
        public string? Description { get; set; }

        [Required]
        public required decimal Price { get; set; }

        [Required]
        public required int Stock { get; set; }

        [Required]
        public required int CategoryId { get; set; }
    }
}

File: frontend_client/lib/model/request/product_request.dart
import 'package:equatable/equatable.dart';

class ProductRequest extends Equatable {
  const ProductRequest({
    this.model,
    this.description,
    this.price,
    this.stock,
    this.categoryId,
    this.productLink,
  });

  factory ProductRequest.fromJson(Map<String, dynamic> json) {
    return ProductRequest(
      model: json['model'] as String?,
      description: json['description'] as String?,
      price: json['price'] as double?,
      stock: json['stock'] as int?,
      categoryId: json['categoryId'] as int?,
      productLink: json['productLink'] as String?,
    );
  }
  final String? model;
  final String? description;
  final double? price;
  final int? stock;
  final int? categoryId;
  final String? productLink;

  Map<String, dynamic> toJson() {
    return {
      'model': model,
      'description': description,
      'price': price,
      'stock': stock,
      'categoryId': categoryId,
      'productLink': productLink,
    };
  }

  @override
  List<Object?> get props => [
    model,
    description,
    price,
    stock,
    categoryId,
    productLink,
  ];
}
